docs/versioning.md

```diff
@@ -0,0 +1,42 @@
+# Kubernetes API and Release Versioning
+
+Legend:
+
+* **Kube &lt;major&gt;.&lt;minor&gt;.&lt;patch&gt;** refers to the version of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.
+* **API vX[betaX]** refers to the version of the HTTP API.
+
+## Release Timeline
+
+### Minor version timeline
+
+* Kube 1.0
+* Kube 1.0.x: We create a 1.0-patch branch and backport critical bugs and security issues to it. Patch releases occur as needed.
+* Kube 1.1-alpha1: Cut from HEAD, smoke tested and released two weeks after Kube 1.0's release. Every two weeks a new alpha is released from HEAD.
+* Kube 1.1-beta1: When HEAD is feature complete, we create a 1.1-snapshot branch and release it as a beta. (The 1.1-snapshot branch may be created earlier if something that definitely won't be in 1.1 needs to be merged to HEAD.) This should occur 6-8 weeks after Kube 1.0. Development continues at HEAD and only fixes are backported to 1.1-snapshot.
+* Kube 1.1-rc1: Released from 1.1-snapshot when it is considered stable and ready for testing. Most users should be able to upgrade to this version in production.
+* Kube 1.1: Final release. Should occur between 3 and 4 months after 1.0.
+
+### Major version timeline
+
+There is no mandated timeline for major versions. They only occur when we need to start the clock on deprecating features. A given major version should be the latest major version for at least one year from its original release date.
+
+## Upgrade policies
+
+* Users can upgrade from any Kube 1.x release to any other Kube 1.x release as a rolling upgrade across their cluster.
+* No hard breaking changes over version boundaries.
+ * For example, if a user is at Kube 1.x, we may require them to upgrade to Kube 1.x+y before upgrading to Kube 2.x. In others words, an upgrade across major versions (e.g. Kube 1.x to Kube 2.x) should effectively be a no-op and as graceful as an upgrade from Kube 1.x to Kube 1.x+1. But you can require someone to go from 1.x to 1.x+y before they go to 2.x.
```

Comment:
2.0 is going to include all v1 APIs, yes. However, 2.0 will not include any v2betaY API's. So if someone had created some v2betaY API objects in 1.x, and only 1.x+y has the full v2 API that they can migrate to, then they need to upgrade to 1.x+y, upgrade their internal objects to v2, and then finally upgrade their cluster to 2.0.

I added a section on "Rationale for API v2 being complete before v2.0's release" to try to clarify this some more.